kratos/solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp

```diff
@@ -246,621 +230,160 @@ class ResidualBasedBossakDisplacementScheme: public Scheme<TSparseSpace,TDenseSp
-            {
-                KRATOS_ERROR << "DISPLACEMENT variable is not allocated for node " << it->Id() << std::endl;
-            }
-            if (it->SolutionStepsDataHas(VELOCITY) == false)
-            {
-                KRATOS_ERROR << "VELOCITY variable is not allocated for node " << it->Id() << std::endl;
-            }
-            if (it->SolutionStepsDataHas(ACCELERATION) == false)
-            {
-                KRATOS_ERROR << "ACCELERATION variable is not allocated for node " << it->Id() << std::endl;
-            }
+        for(auto& rnode : rModelPart.Nodes()) {
+            KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(DISPLACEMENT,rnode) 
+            KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VELOCITY,rnode) 
+            KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(ACCELERATION,rnode) 
+    
+            KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_X, rnode) 
+            KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Y, rnode) 
+            KRATOS_CHECK_DOF_IN_NODE(DISPLACEMENT_Z, rnode) 
         }
 
-        // Check that dofs exist
-        for(ModelPart::NodesContainerType::iterator it=rModelPart.NodesBegin();
-                it!=rModelPart.NodesEnd(); it++)
-        {
-            if(it->HasDofFor(DISPLACEMENT_X) == false)
-            {
-                KRATOS_ERROR << "missing DISPLACEMENT_X dof on node " << it->Id() << std::endl;
-            }
-            if(it->HasDofFor(DISPLACEMENT_Y) == false)
-            {
-                KRATOS_ERROR << "missing DISPLACEMENT_Y dof on node " << it->Id() << std::endl;
-            }
-            if(it->HasDofFor(DISPLACEMENT_Z) == false)
-            {
-                KRATOS_ERROR << "missing DISPLACEMENT_Z dof on node " << it->Id() << std::endl;
-            }
-        }
+        // Check for minimum value of the buffer index
+        // Verify buffer size
+        KRATOS_ERROR_IF(rModelPart.GetBufferSize() < 2) << "Insufficient buffer size. Buffer size should be greater than 2. Current size is" << rModelPart.GetBufferSize() << std::endl;
 
         // Check for admissible value of the AlphaBossak
-        if(mAlpha.m > 0.0 || mAlpha.m < -0.3)
-        {
-            KRATOS_ERROR << "Value not admissible for AlphaBossak. Admissible values should be between 0.0 and -0.3. Current value is " << mAlpha.m << std::endl;
-        }
+        KRATOS_ERROR_IF(mAlpha.m > 0.0 || mAlpha.m < -0.3) << "Value not admissible for AlphaBossak. Admissible values should be between 0.0 and -0.3. Current value is " << mAlpha.m << std::endl;
 
         // Check for minimum value of the buffer index
         // Verify buffer size
-        if (rModelPart.GetBufferSize() < 2)
-        {
-            KRATOS_ERROR << "insufficient buffer size. Buffer size should be greater than 2. Current size is" << rModelPart.GetBufferSize() << std::endl;
-        }
+        KRATOS_ERROR_IF(rModelPart.GetBufferSize() < 2) << "insufficient buffer size. Buffer size should be greater than 2. Current size is" << rModelPart.GetBufferSize() << std::endl;
```

Comment:
This is redundant right?

Reply:
I changed it, not duplciate it

Reply:
I mean there appear two statements (lines 381 and 388) that check the same thing.

Reply:
True, done